server.go

```diff
@@ -22,6 +45,25 @@ type Server struct {
 	jq  *JobQueue
 	pq  *ProjectQueue
 	cfg *Config
+
+	// Events are pushed to this channel.
+	Notifier chan *C
+
+	// Channel for adding new client connections.
+	newClients chan *C
+
+	// Channel for signaling a closed client connection.
+	closingClients chan *C
+
+	// Client connections registry.
+	//
+	// Cs sent to the newClients channel are being added to the registry.
+	// We use a reference to the C struct so that the conncetions can be
+	// uniquely identified for deletions.
+	clients map[*C]bool
+
+	// A connection queue used to track all open connections by their id.
+	cq map[string]bool
```

Comment:
All these fields and methods could go into a separate struct, ideally extracted into its own package and exporting its http handlers that one can use selectively.

Reply:
Extracted into a `broker` package.